src/smtml/expr.ml

```diff
@@ -424,7 +425,9 @@ let rec relop ty op hte1 hte2 =
     if Int32.equal b1 b2 then relop Ty_bool Eq os1 os2 else value False
   | Ne, Ptr { base = b1; offset = os1 }, Ptr { base = b2; offset = os2 } ->
     if Int32.equal b1 b2 then relop Ty_bool Ne os1 os2 else value True
-  | ( (LtU | LeU | GtU | GeU)
+  | GtU, _, _ -> relop ty LtU hte2 hte1
+  | GeU, _, _ -> relop ty LeU hte2 hte1
```

Comment:
```suggestion
  | Gt, _, _ -> relop ty Lt hte2 hte1
  | GtU, _, _ -> relop ty LtU hte2 hte1
  | Ge, _, _ -> relop ty Le hte2 hte1
  | GeU, _, _ -> relop ty LeU hte2 hte1
``` 
It's missing for signed operations as well. 

But here you should leave the pattern match for the pointer compassion as is, because it will be resolved concretely. And, just to be safe, add these rules just before the default case: `| _, _, _ -> raw_relop ty op hte1 hte2` on line 446.